### PR TITLE
Create Categories on-the-fly on Link edit form

### PR DESCRIPTION
**Pull request by alikon**
#### Summary of Changes
Create Categories on-the-fly on Weblink edit form

#### Testing Instructions
Apply patch
Go to Administrator -> Components -> Weblinks ->Link
Click "New"
Enter link details
on category list, enter a new category name and press "Enter"  
(work like the cms feature for content,contact, ect)

### CODE REVIEW

File: src/administrator/components/com_weblinks/models/forms/weblink.xml [hunk 1]
```diff
@@ -17,11 +17,14 @@
...
-		</field>
-
+		<field	name="catid"
+			type="categoryedit"
+			extension="com_weblinks"
+			label="JCATEGORY"
+			description="COM_WEBLINKS_FIELD_CATEGORY_DESC"
+			required="true"
+			allowAdd="true"
+			default="" />
```
Comment (yvesh) [inline src/administrator/components/com_weblinks/models/forms/weblink.xml]: Code style (Yes i know the whole file is a mess :-))

**Comment by yvesh** [comment]
I have tested this item :white_check_mark: successfully on 6e425b7d5f5543f4fcdb837bc7aa7c86fff0f00a<br /><br />Works as expected. Thank you Nicola for your work :-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/252">issues.joomla.org/weblinks/252</a>.</sub>

**Comment by alikon** [comment]
what is supposed to be the correct CS ? 
... i always fight with CS/travis etc..... cause i'm lazy :stuck_out_tongue:

**Comment by yvesh** [comment]
@alikon http://joomla.github.io/coding-standards/?coding-standards/chapters/xml.md

So in this case

```xml
<field
	name="catid"
	type="categoryedit"
	extension="com_weblinks"
	label="JCATEGORY"
	description="COM_WEBLINKS_FIELD_CATEGORY_DESC"
	required="true"
	allowAdd="true"
	default=""
 />
```

**Comment by roland-d** [comment]
I have tested this item :white_check_mark: successfully on 6e425b7d5f5543f4fcdb837bc7aa7c86fff0f00a<br /><br />After applying the patch, I am able to add categories when editing/creating a weblink.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/252">issues.joomla.org/weblinks/252</a>.</sub>

**Comment by zero-24** [comment]
I guess we should implement ACL checks like: https://github.com/joomla/joomla-cms/pull/11244

**Comment by yvesh** [comment]
@zero-24 But this is maybe something for a second PR imo.

Any ideas how to implement that without duplicate code? That model check should be either in a helper or move into the JModelAdmin itself. What do you think? Else we will have it in so many areas..

**Comment by zero-24** [comment]
ACL: https://github.com/alikon/weblinks/pull/4

**Comment by alikon** [comment]
@zero-24  thanks
merged

**Comment by zero-24** [comment]
:+1:

**Comment by zero-24** [comment]
I think after the last changes we can go with the ACL fixes?

**Comment by alikon** [comment]
@zero-24  can you be more detailed ?
if i'm not missing something, and this is probable, https://github.com/joomla/joomla-cms/pull/11244 are in place....

**Comment by zero-24** [comment]
Yes. I want to say we should merge here ;)

File: src/administrator/components/com_weblinks/models/forms/weblink.xml [hunk 2]
```diff
@@ -17,11 +17,14 @@
 			description="COM_WEBLINKS_FIELD_ALIAS_DESC"
 			hint="JFIELD_ALIAS_PLACEHOLDER" />
 
-		<field name="catid" type="categoryedit" extension="com_weblinks"
-			label="JCATEGORY" description="COM_WEBLINKS_FIELD_CATEGORY_DESC"
-		>
-		</field>
-
+		<field	name="catid"
```
Comment (810) [inline src/administrator/components/com_weblinks/models/forms/weblink.xml]: please use 1 space here for codestyle

Comment (zero-24) [inline src/administrator/components/com_weblinks/models/forms/weblink.xml]: You can edit this if this via github if you want :)

Comment (alikon) [inline src/administrator/components/com_weblinks/models/forms/weblink.xml]: done

**Comment by chrisdavenport** [comment]
I have tested this item :white_check_mark: successfully on ee124c6695020311e2d6093473d9842d0512d45a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/252">issues.joomla.org/tracker/weblinks/252</a>.</sub>

**Pull request merged**